src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -132,9 +170,33 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 expressionOpt,
                 iDisposableConversion,
                 boundBody,
+                disposeMethodOpt: null, // This ensures that a pattern-matched Dispose statement is not used.
                 hasErrors);
         }
 
+        /// <summary>
+        /// Checks for a Dispose method on exprType. Failing to satisfy the pattern is not an error -
+        /// it just means we have to check for an interface instead.
+        /// </summary>
+        /// <param name="exprType">Type of the expression over which to iterate</param>
+        /// <param name="diagnostics">Populated with warnings if there are near misses</param>
+        /// <returns>True if a matching method is found (still need to verify return type).</returns>
+        private MethodSymbol TryFindDisposePattern(TypeSymbol exprType, DiagnosticBag diagnostics)
+        {
+            LookupResult lookupResult = LookupResult.GetInstance();
+            SyntaxNode exp = _syntax.Expression != null ? (SyntaxNode) _syntax.Expression : (SyntaxNode) _syntax.Declaration;
```

Comment:
I would just call this `syntax`. `exp` implies it is an expression, but a declaration is not an expression.